### PR TITLE
Keyword axioms

### PR DESCRIPTION
**Pull request by aclarkxyz**
Added "keyword" option to axiom rules

### CODE REVIEW

File: src/com/cdd/bao/axioms/AxiomVocab.java [hunk 1]
```diff
@@ -64,8 +64,8 @@ public static Type valueOf(int rawVal)
 
 	public static class Term
 	{
-		public String valueURI;
-		public boolean wholeBranch;
+		public String valueURI = null;
+		public boolean wholeBranch = false;
```
Comment (gedeck) [inline src/com/cdd/bao/axioms/AxiomVocab.java]: These are the default values.

File: src/com/cdd/bao/axioms/AxiomVocab.java [hunk 2]
```diff
@@ -95,31 +95,77 @@ public int hashCode()
 		}
 	}
 	
+	public static class Keyword
+	{
+		public String text = null; // short string that must be present, and separated by a boundary
```
Comment (gedeck) [inline src/com/cdd/bao/axioms/AxiomVocab.java]: What boundary?

File: src/com/cdd/bao/axioms/AxiomVocab.java [hunk 3]
```diff
@@ -172,6 +220,14 @@ public boolean equals(Object obj)
...
+			{
+				if (other.keyword != null) return false;
+			}
+			else
+			{
+				if (!keyword.equals(other.keyword)) return false;
+			}
 			int sz = ArrayUtils.getLength(impact);
 			if (sz != ArrayUtils.getLength(other.impact)) return false;
 			for (int n = 0; n < sz; n++) if (!impact[n].equals(other.impact[n])) return false;
```
Comment (gedeck) [inline src/com/cdd/bao/axioms/AxiomVocab.java]: I think this is overly complex. 
```
return type == other.type && saveEquals(subject, other.subject) && saveEquals(keyword, other.keyword) &&
  saveEqualsArray(impact, other.impact);
```
Methods in `InitParams`

Comment (gedeck) [reply src/com/cdd/bao/axioms/AxiomVocab.java]: Keyword not included in hashCode
```
@Override
public int hashCode()
{
  return Objects.hash(subject, impact && Arrays.asList(impact), keyword.hashCode());
}
```

File: src/com/cdd/bao/axioms/AxiomVocab.java (hunk 1 above)
Comment (aclarkxyz) [reply src/com/cdd/bao/axioms/AxiomVocab.java]: Setting to null is a "clue" that the constructor is likely to just leave it that way... whereas leaving it un-initialised implies that the constructor is going to make the call explicitly.

**Pull request merged**